File: typescript/src/pricecalculator.ts
class PriceCalculator {
    calculatePrice(article: any, customerStatus: any): any {
        if (article.name() === "Smartphone") {
            return 599.99;
        } else if (article.name() === "Coffee Maker") {
            return 49.99;
        } else if (article.name() === "Running Shoes") {
            return 89.95;
        } else if (article.name() === "Wireless Headphones") {
            return 129.99;
        } else if (article.name() === "Laptop") {
            return 999.99;
        } else if (article.name() === "Yoga Mat") {
            return 19.99;
        } else if (article.name() === "Sunglasses") {
            return 59.95;
        } else if (article.name() === "Backpack") {
            return 39.99;
        } else if (article.name() === "Portable Speaker") {
            return 79.99;
        } else if (article.name() === "Fancy Watch") {
            return 199.99;
        } else {
            throw new Error('No price for article: ' + article.name());
        }
    }
}

export { PriceCalculator };

File: java/src/main/java/com/innoq/automatedtesting/samples/shoppingcart/PriceCalculator.java
package com.innoq.automatedtesting.samples.shoppingcart;

import java.math.BigDecimal;

public class PriceCalculator {
    public BigDecimal calculatePrice(Article article, CustomerStatus customerStatus) {
        BigDecimal originalPrice;

        switch (article.name()) {
            case "Smartphone":
                originalPrice = new BigDecimal("599.99");
                break;
            case "Coffee Maker":
                originalPrice = new BigDecimal("49.99");
                break;
            case "Running Shoes":
                originalPrice = new BigDecimal("89.95");
                break;
            case "Wireless Headphones":
                originalPrice = new BigDecimal("129.99");
                break;
            case "Laptop":
                originalPrice = new BigDecimal("999.99");
                break;
            case "Yoga Mat":
                originalPrice = new BigDecimal("19.99");
                break;
            case "Sunglasses":
                originalPrice = new BigDecimal("59.95");
                break;
            case "Backpack":
                originalPrice = new BigDecimal("39.99");
                break;
            case "Portable Speaker":
                originalPrice = new BigDecimal("79.99");
                break;
            case "Fancy Watch":
                originalPrice = new BigDecimal("199.99");
                break;
            default:
                throw new RuntimeException("No price for article: " + article.name());
        }

        BigDecimal discountPercentage;
        switch (customerStatus) {
            case REGULAR:
                discountPercentage = new BigDecimal("0.05"); // 5% discount
                break;
            case SILVER:
                discountPercentage = new BigDecimal("0.10"); // 10% discount
                break;
            case GOLD:
                discountPercentage = new BigDecimal("0.15"); // 15% discount
                break;
            case PLATINUM:
                discountPercentage = new BigDecimal("0.20"); // 20% discount
                break;
            default:
                discountPercentage = BigDecimal.ZERO; // No discount
        }

        BigDecimal discountAmount = originalPrice.multiply(discountPercentage);
        return originalPrice.subtract(discountAmount);
    }
}
